pkg/types/validation/installconfig.go

```diff
@@ -596,15 +596,25 @@ func validateNetworkingClusterNetworkMTU(c *types.InstallConfig, fldPath *field.
 	return allErrs
 }
 
-func validateControlPlane(platform *types.Platform, pool *types.MachinePool, fldPath *field.Path) field.ErrorList {
+func validateControlPlane(platform *types.Platform, pool *types.MachinePool, isSingleNodeOpenShift bool, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	if pool.Name != types.MachinePoolControlPlaneRoleName {
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("name"), pool.Name, []string{types.MachinePoolControlPlaneRoleName}))
 	}
-	if pool.Replicas != nil && *pool.Replicas == 0 {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("replicas"), pool.Replicas, "number of control plane replicas must be positive"))
+	if pool.Replicas == nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("replicas"), pool.Replicas, "number of control plane replicas must be provided"))
+	} else {
+		switch {
+		case *pool.Replicas == 0:
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("replicas"), pool.Replicas, "number of control plane replicas must be positive"))
+		case *pool.Replicas == 1 && !supportedSingleNodePlatform(isSingleNodeOpenShift, platform):
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("replicas"), pool.Replicas, "Single Node OpenShift(SNO) not supported by this install method"))
+		case *pool.Replicas < 3:
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("replicas"), pool.Replicas, "number of control plane replicas must be >= 3"))
+		default:
+			allErrs = append(allErrs, ValidateMachinePool(platform, pool, fldPath)...)
+		}
 	}
-	allErrs = append(allErrs, ValidateMachinePool(platform, pool, fldPath)...)
```

Comment:
@zaneb , @patrickdillon I am changing the behavior here. ValidateMachinePool() was called even when number of replicas were incorrect. Now, it is called only when the number of control plane replicas is correct. Do you see an issue with that?